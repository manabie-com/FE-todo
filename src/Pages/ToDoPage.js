/* eslint-disable react-hooks/exhaustive-deps */
import React from "react";

import { AuthButtom, BaseInput, TodosToolBar, TodoItems } from "../Comps";
import { loginPath } from "../constant";
import Service from "../service";
import Context from "../store";
import { devidedTodos } from "../utils";

const ToDoPage = ({ history }) => {
  /**
   * get Store using Context
   */
  const propsContext = React.useContext(Context);
  const {
    state: { todos = [], content = "" },
    onChangeContent = () => {},
    createTodo = () => {},
    setTodos = () => {},
  } = propsContext;

  /**
   *
   * logics provided to create todo
   */
  const onCreateTodo = async (e) => {
    if (
      e.key === "Enter" &&
      typeof content === "string" &&
      content.length !== 0
    ) {
      try {
        const resp = await Service.createTodo(content);
        createTodo(resp);
      } catch (e) {
        if (e.response.status === 401) {
          history.push(loginPath);
        }
      }
    }
  };

  /**
   * logics provied to get data
   */
  const getData = async () => {
    const resp = await Service.getTodos();
    setTodos(resp || []);
  };

  /**
   * effect call data after first time render
   */
  React.useEffect(() => {
    getData();
  }, []);

  /**
   * Save todos in Local Store on Browser
   */
  React.useEffect(() => {
    localStorage.setItem("todoData", JSON.stringify(todos));
  }, [todos]);

  const { todoActive = [], todoCompleted = [] } = devidedTodos(
    /* function prduces 2 array base on status */ todos
  );

  return (
    <div className="container">
      {/* Logout Btn */}
      <AuthButtom auth="logout" />
      {/* Input to Create Todo */}
      <BaseInput
        placeholder="What need to be done?"
        value={/* main content on Create */ content}
        onChange={(e) => {
          onChangeContent(e.target.value);
        }}
        onKeyDown={/* support save when user enter */ onCreateTodo}
      />
      {/* Toolbar go with todos list */}
      <TodosToolBar />
      {/* todo-lists */}
      <div>
        {/* todos are Actives show first  */}
        {todoActive.map((todo, index) => {
          return <TodoItems key={index} {...todo} />;
        })}
        {/* todos are Completed show last  */}
        {todoCompleted.map((todo, index) => {
          return <TodoItems key={index} {...todo} />;
        })}
      </div>
    </div>
  );
};

export default ToDoPage;
