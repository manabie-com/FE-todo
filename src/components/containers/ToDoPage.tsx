import React, { useEffect, useReducer, useState } from "react";
import { RouteComponentProps } from "react-router-dom";
import reducer, { initialState } from "../../store/reducer";
import {
  setTodos,
  createTodo,
  deleteTodo,
  deleteAllTodos,
  updateTodoStatus,
  toggleAllTodos,
} from "../../store/actions";
import { isTodoCompleted } from "../../utils";
import Service from "../../service";
import { TodoStatus, Todo } from "../../models/todo";
import { Card, Popup, Input, Sort } from "../UI/index";
type EnhanceTodoStatus = TodoStatus | "ALL";

const ToDoPage = ({ history }: RouteComponentProps) => {
  const [{ todos }, dispatch] = useReducer(reducer, initialState);
  const [showing, setShowing] = useState<EnhanceTodoStatus>("ALL");
  const [togglePopup, setTogglePopup] = useState<boolean>(false);
  const [todoEdit, setTodoEdit] = useState<Todo | null>(null);

  useEffect(() => {
    (async () => {
      const resp = (await Service.getTodos()) ?? [];
      dispatch(setTodos(resp || []));
    })();
  }, [togglePopup]);

  const onCreateTodo = async (
    e: React.KeyboardEvent<HTMLInputElement>,
    inputRef: any
  ) => {
    if (e.key === "Enter" && inputRef.current) {
      try {
        const resp = await Service.createTodo(inputRef.current.value);
        dispatch(createTodo(resp));
        inputRef.current.value = "";
      } catch (e) {
        if (e.response.status === 401) {
          history.push("/");
        }
      }
    }
  };

  const onToggleAllTodo = (e: React.ChangeEvent<HTMLInputElement>) => {
    dispatch(toggleAllTodos(e.target.checked));
  };

  const onDeleteAllTodo = () => {
    dispatch(deleteAllTodos());
  };

  const onTogglePopup = (event: any, todo: Todo) => {
    if (todo) {
      setTodoEdit(todo);
    }
    setTogglePopup(!togglePopup);
  };

  const showTodos = todos.filter((todo) => {
    switch (showing) {
      case TodoStatus.ACTIVE:
        return todo.status === TodoStatus.ACTIVE;
      case TodoStatus.COMPLETED:
        return todo.status === TodoStatus.COMPLETED;
      default:
        return true;
    }
  });

  const activeTodos = todos.reduce(function (accum, todo) {
    return isTodoCompleted(todo) ? accum : accum + 1;
  }, 0);

  const onUpdateTodoStatus = (
    e: React.ChangeEvent<HTMLInputElement>,
    todoId: string
  ) => {
    dispatch(updateTodoStatus(todoId, e.target.checked));
  };

  const deleteTodoFunc = (todoId: string) => {
    dispatch(deleteTodo(todoId));
  };

  return (
    <div className="ToDo__container">
      <h1>Manabie Todo</h1>
      <Popup
        todolist={showTodos}
        isToggle={togglePopup}
        onTogglePopup={onTogglePopup}
        todoEdit={todoEdit}
      />
      <Input onCreateTodo={onCreateTodo} />
      <Sort
        showing={showing}
        setShowing={setShowing}
        onDeleteAllTodo={onDeleteAllTodo}
        onToggleAllTodo={onToggleAllTodo}
        activeTodos={activeTodos}
      />
      <Card
        todoList={showTodos}
        onUpdateTodoStatus={onUpdateTodoStatus}
        deleteTodo={deleteTodoFunc}
        onTogglePopup={onTogglePopup}
      />
    </div>
  );
};

export default ToDoPage;
