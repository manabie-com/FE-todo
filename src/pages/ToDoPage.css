.ToDo__container {
    border: 1px solid rgba(0,0,0, 0.13);
    border-radius: 8px;
    width: 500px;
    margin-top: 5rem;
    padding: 24px;
}

.Todo__creation {
    display: flex;
}

.Todo__input {
    flex: 1 1;
}

.ToDo__list {
    display: flex;
    flex-direction: column;
    margin-top: 1.5rem;
}

.Todo__toolbar {
    display: flex;
    margin-top: 24px;
    align-items: center;
}

.Todo__tabs {
    display: flex;
    justify-content: flex-end;
    flex-grow: 1;
}

.Todo__tabs > *:not(:last-child) {
    margin-right: 8px;
}

.Action__btn.active {
    box-shadow: inset 2px 2px #cecece;
}
