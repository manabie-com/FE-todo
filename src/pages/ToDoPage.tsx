import React, { useEffect, useReducer, useRef, useState } from "react";
import { RouteComponentProps } from "react-router-dom";

import reducer, { initialState } from "../store/reducer";
import {
  setTodos,
  createTodo,
  deleteTodo,
  toggleAllTodos,
  deleteAllTodos,
  updateTodoStatus,
  updateTodo,
} from "../store/actions";
import Service from "../service";
import { Todo, TodoStatus } from "../models/todo";
import { isTodoCompleted } from "../utils";
import TodoItem from "../components/TodoItem";
import ActionButton from "../components/ActionButton";
import styled from "styled-components";

const Container = styled.div`
  padding-top: 5rem;
  padding-left: 30px;
  padding-right: 30px;
  display: flex;
  justify-content: center;
  width: 100%;

  .todo {
    max-width: 500px;
    width: 100%;

    &__page-title {
      font-size: 30px;
      text-align: left;
      padding-bottom: 24px;
      font-weight: bold;
    }

    &__creation {
      display: flex;

      &__input {
        flex: 1 1;
      }
    }

    &__list {
      &-wrapper {
        border: 1px solid rgba(0, 0, 0, 0.13);
        border-radius: 8px;
        margin-top: 30px;
        box-shadow: 0 0 5px rgb(0 0 0 / 20%);
      }
    }

    &__toolbar {
      display: flex;
      justify-content: space-between;
      padding: 8px 15px;
      flex-wrap: wrap;

      input[type="checkbox"] {
        min-height: unset;
      }
    }

    &__tabs {
      display: flex;
      justify-content: center;

      > *:not(:last-child) {
        margin-right: 8px;
      }
    }

    &__clear {
      display: flex;
      align-items: center;
      justify-content: center;
    }

    &__empty-block {
      border-bottom: 1px solid lightgrey;
      padding: 21px 15px;
      font-style: italic;
      color: lightgray;
    }
  }

  @media screen and (max-width: 567px) {
    .todo {
      &__clear {
        padding-top: 10px;
        width: 100%;
      }
    }
  }
`;

type EnhanceTodoStatus = TodoStatus | "ALL";

const ToDoPage = ({ history }: RouteComponentProps) => {
  const [{ todos }, dispatch] = useReducer(reducer, initialState);
  // change showing to selectedTab for meaning
  const [selectedTab, setSelectedTab] = useState<EnhanceTodoStatus>("ALL");
  const inputRef = useRef<HTMLInputElement>(null);

  const listTabs: EnhanceTodoStatus[] = [
    "ALL",
    TodoStatus.ACTIVE,
    TodoStatus.COMPLETED,
  ];

  useEffect(() => {
    (async () => {
      const resp = await Service.getTodos();
      dispatch(setTodos(resp || []));
    })();
  }, []);

  const onCreateTodo = async (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === "Enter" && inputRef.current) {
      try {
        const content = inputRef?.current?.value?.trim();
        if (content === "") return;
        const resp = await Service.createTodo(content);
        inputRef.current.value = "";
        dispatch(createTodo(resp));
      } catch (e) {
        if (e.response.status === 401) {
          history.push("/");
        }
      }
    }
  };

  const onUpdateTodoStatus = (
    e: React.ChangeEvent<HTMLInputElement>,
    todoId: string
  ) => {
    dispatch(updateTodoStatus(todoId, e.target.checked));
  };

  const onToggleAllTodo = (e: React.ChangeEvent<HTMLInputElement>) => {
    dispatch(toggleAllTodos(e.target.checked));
  };

  const onDeleteAllTodo = () => {
    dispatch(deleteAllTodos());
  };

  const handleDeleteTodo = (todoId: string) => {
    dispatch(deleteTodo(todoId));
  };

  const handleUpdateTodoItem = (todo: Todo) => {
    dispatch(updateTodo(todo));
  };

  const showTodos = todos.filter((todo) => {
    switch (selectedTab) {
      case TodoStatus.ACTIVE:
        return todo.status === TodoStatus.ACTIVE;
      case TodoStatus.COMPLETED:
        return todo.status === TodoStatus.COMPLETED;
      default:
        return true;
    }
  });

  const activeTodos = todos.reduce(function (accum, todo) {
    return isTodoCompleted(todo.status) ? accum : accum + 1;
  }, 0);

  return (
    <Container>
      <div className="todo">
        <div className="todo__page-title">Todo List</div>
        <div className="todo__creation">
          <input
            ref={inputRef}
            className="todo__creation__input"
            placeholder="What need to be done?"
            onKeyDown={onCreateTodo}
          />
        </div>
        <div className="todo__list-wrapper">
          {showTodos.length === 0 ? (
            <div className="todo__empty-block">Empty List</div>
          ) : (
            ""
          )}
          <div className="todo__list">
            {showTodos.map((todo, index) => {
              return (
                <TodoItem
                  key={todo.id}
                  todo={todo}
                  onUpdate={(e) => onUpdateTodoStatus(e, todo.id)}
                  onDelete={() => handleDeleteTodo(todo.id)}
                  onUpdateContent={(content: string) =>
                    handleUpdateTodoItem({ ...todo, content })
                  }
                />
              );
            })}
          </div>
          <div className="todo__toolbar">
            {todos.length > 0 ? (
              <input
                type="checkbox"
                checked={activeTodos === 0}
                onChange={onToggleAllTodo}
              />
            ) : (
              <div />
            )}
            <div className="todo__tabs">
              {listTabs.map((status) => (
                <ActionButton
                  key={status}
                  text={status.toLowerCase()}
                  active={selectedTab === status}
                  onClick={() => setSelectedTab(status)}
                />
              ))}
            </div>
            <div className="todo__clear">
              <ActionButton text="Clear all todos" onClick={onDeleteAllTodo} />
            </div>
          </div>
        </div>
      </div>
    </Container>
  );
};

export default ToDoPage;
