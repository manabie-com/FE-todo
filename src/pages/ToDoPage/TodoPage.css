.todo{
  background: linear-gradient(90deg, rgba(2,0,36,1) 0%, rgba(108,9,121,0.8715861344537815) 38%, rgba(0,212,255,1) 98%, rgba(0,212,255,1) 98%);
  background-size: cover;
  max-height: 9999px;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
}
.todo__heading{
  margin: 0;
  color: white;
  font-size: 4rem;
  margin: 3rem auto;
}
.todo__container {
  border: 1px solid rgba(0,0,0, 0.13);
  border-radius: 8px;
  max-width: 500px;
  padding: 2.4rem;
  box-shadow: 2px 2px 1px rgba(0,0,0, 0.09),
              3px 2px 3px rgba(0,0,0, 0.05);
  background-color: whitesmoke;
  margin: 2rem 1rem;
}


