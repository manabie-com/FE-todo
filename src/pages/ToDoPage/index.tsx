import React, { useEffect, useState, useRef } from 'react';
import { RouteComponentProps } from 'react-router-dom';
import {StateInspector, useReducer} from 'reinspect'

import {Todo} from '../../models/todo';

import reducer, { initialState } from '../../store/reducer';
import {
  setTodos,
  createTodo,
  deleteTodo,
  toggleAllTodos,
  deleteAllTodos,
  updateTodoStatus,
  updateTodo
} from '../../store/actions';
import Service from '../../service';
import { TodoStatus } from '../../models/todo';
import { isTodoCompleted } from '../../utils';

import './index.css';
import TodoItem from '../../components/TodoItem';

type EnhanceTodoStatus = TodoStatus | 'ALL';

const ToDoPage = ({ history }: RouteComponentProps) => {
  const [{ todos }, dispatch] = useReducer(reducer, initialState, (a) => a, 'todosState');
  const [todoInput, setTodoInput] = useState('');
  const [showing, setShowing] = useState<EnhanceTodoStatus>('ALL');
  const [editId, setEditId] = useState('');
  const inputRef = useRef<HTMLInputElement>(null);

  useEffect(() => {
    (async () => {
      const resp = await Service.getTodos();

      console.log('RUN', resp);
      dispatch(setTodos(resp || []));
    })();
  }, []);

  const updateEditId = (id: string) => {
    setEditId(id)
  }

  const onCreateTodo = async (e: React.KeyboardEvent<HTMLInputElement>) => {
    const inputValue = todoInput.trim();
    if (e.key === 'Enter' && inputValue.length > 0) {
      try {
        const resp = await Service.createTodo(inputValue);
        dispatch(createTodo(resp));

        // const todoo = {
        //   content: 'asd',
        //   user_id: 'user_id',
        //   id: 'fine',
        //   status: 'ACTIVE',
        //   created_date: '2021-01-30T14:40:29.202Z'
        // } as Todo;
        // dispatch(createTodo(todoo));
        
        // Clear todo input
        setTodoInput('');
      } catch (e) {
        if (e.response.status === 401) {
          history.push('/');
        }
      }
    }
  };

  const onToggleAllTodo = (e: React.ChangeEvent<HTMLInputElement>) => {
    dispatch(toggleAllTodos(e.target.checked));
  };

  const onDeleteAllTodo = () => {
    dispatch(deleteAllTodos());
  };

  const showTodos = todos
    .filter((todo) => {
      switch (showing) {
        case TodoStatus.ACTIVE:
          return todo.status === TodoStatus.ACTIVE;
        case TodoStatus.COMPLETED:
          return todo.status === TodoStatus.COMPLETED;
        default:
          return true;
      }
    })
    .sort((a, b) => (a.created_date > b.created_date ? -1 : 1)); // sort by recent time order

  const activeTodos = todos.reduce(function (accum, todo) {
    return isTodoCompleted(todo) ? accum : accum + 1;
  }, 0);

  const onTodoInput = (e: React.ChangeEvent<HTMLInputElement>) => {
    setTodoInput(e.currentTarget.value);
  };

  // const handleEdit = (id: string) => {
  //   // setTodoInput(e.currentTarget.value);
  //   console.log(id);
  //   setEditId(id);
  //   //    dispatch(updateTodo(id, ))
  // };

  return (
    <div className='ToDo__container'>
      <div className='Todo__creation'>
        <input
          ref={inputRef}
          value={todoInput}
          className='Todo__input'
          placeholder='What need to be done?'
          onKeyDown={onCreateTodo}
          onChange={onTodoInput}
        />
      </div>
      <div className='ToDo__list'>
        {showTodos.map((todo) => {
          const {id = '0'} = todo;
          return <TodoItem todo={todo} updateEditId={updateEditId} edit={id === editId}/>;
          //   const { id = '123', content = '' } = todo;
          //   return (
          //     <div key={id} className='ToDo__item'>
          //       <input
          //         type='checkbox'
          //         checked={isTodoCompleted(todo)}
          //         onChange={(e) => onUpdateTodoStatus(e, id)}
          //       />

          //       <input className='Todo__content' value={content} disabled></input>

          //       <div
          //         className='Todo__delete'
          //         onClick={() => dispatch(deleteTodo(id))}
          //       >
          //         <i className='bx bx-x icon'></i>
          //       </div>

          //       <div className='Todo__edit' onClick={() => handleEdit(id)}>
          //         <i className='bx bxs-pencil icon'></i>
          //       </div>
          //     </div>
          //   );
        })}
      </div>
      <div className='Todo__toolbar'>
        {todos.length > 0 ? (
          <input
            type='checkbox'
            checked={activeTodos === 0}
            onChange={onToggleAllTodo}
          />
        ) : (
          <div />
        )}
        <div className='Todo__tabs'>
          <button className='Action__btn' onClick={() => setShowing('ALL')}>
            All
          </button>
          <button
            className='Action__btn'
            onClick={() => setShowing(TodoStatus.ACTIVE)}
          >
            Active
          </button>
          <button
            className='Action__btn'
            onClick={() => setShowing(TodoStatus.COMPLETED)}
          >
            Completed
          </button>
        </div>
        <button className='Action__btn' onClick={onDeleteAllTodo}>
          Clear all todos
        </button>
      </div>
    </div>
  );
};

export default ToDoPage;
