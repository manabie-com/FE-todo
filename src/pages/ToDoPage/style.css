.TodoPage{
    display: -webkit-box;
    display: -moz-box;
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;

    -webkit-justify-content: center;
    justify-content: center;

    -webkit-box-align: center;
    -moz-box-align: center;
    -ms-flex-align: center;
    -webkit-align-items: center;
    align-items: center;

    width:100%;
    min-height: 100vh;

    background: #fafafa;
}

.ToDo__container{
    margin:50px 0;
    background: #ffffff;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 3px rgba(0, 0, 0, 0.24);

    -webkit-border-radius: 10px;
    -moz-border-radius: 10px;
    -o-border-radius: 10px;
    -ms-border-radius: 10px;
    border-radius: 10px;
}

.Todo__tabs{
    margin-right:30px;
}