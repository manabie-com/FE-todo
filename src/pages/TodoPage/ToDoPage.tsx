import React, { useRef } from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { useQueryClient } from 'react-query';
import { useDispatch } from 'react-redux';

import { TodoStatus, Todo } from '../../models/todo';
import { isTodoCompleted } from '../../utils';
import {
  useTodos,
  useCreateTodo,
  useDeleteTodo,
  useUpdateTodo,
  useDeleteTodos,
  useCompleteTodoAll,
  useActiveAll
} from 'hooks/todos';
import BaseButton from 'components/BaseButton';
import SingleTodo from './Todo';

import TodoPageWrapper from './style';
import useLocalStorage from 'hooks/useLocalStorage';
import { clearLogin } from 'store/modules/auth/slice';

const ToDoPage = ({ history }: RouteComponentProps) => {
  const inputRef = useRef<HTMLInputElement>(null);
  const queryClient = useQueryClient();
  const [, setToken] = useLocalStorage('token', '');
  const dispatch = useDispatch();

  const { todos } = useTodos();
  const { createTodo, isCreating } = useCreateTodo();
  const { deleteTodo } = useDeleteTodo();
  const { deleteTodos } = useDeleteTodos();
  const { updateTodo } = useUpdateTodo();
  const { completeAll } = useCompleteTodoAll();
  const { activeAll } = useActiveAll();

  const handleDeleteTodo = (id: number) => {
    deleteTodo(id, {
      onSuccess: () => {
        queryClient.invalidateQueries('todos');
      }
    });
  };

  const onCreateTodo = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter' && inputRef.current && !isCreating) {
      if (inputRef.current.value) {
        createTodo(inputRef.current.value, {
          onSuccess: () => {
            queryClient.invalidateQueries('todos');
            inputRef.current.value = '';
          },
          onError: () => history.push('/')
        });
      }
    }
  };

  const handleCompleteAll = () => {
    completeAll(undefined, {
      onSuccess: () => {
        queryClient.invalidateQueries('todos');
      }
    });
  };

  const handleActiveAll = () => {
    activeAll(undefined, {
      onSuccess: () => {
        queryClient.invalidateQueries('todos');
      }
    });
  };

  const handleUpdateStatus = (
    e: React.ChangeEvent<HTMLInputElement>,
    todo: Todo
  ) => {
    todo.status = e.target.checked ? TodoStatus.COMPLETED : TodoStatus.ACTIVE;
    updateTodo(todo, {
      onSuccess: () => {
        queryClient.invalidateQueries('todos');
      }
    });
  };

  const onToggleAllTodo = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (!e.target.checked) {
      handleActiveAll();
    } else {
      handleCompleteAll();
    }
  };

  const handleDeleteAll = () => {
    deleteTodos(undefined, {
      onSuccess: () => queryClient.invalidateQueries('todos')
    });
  };

  const handleLogout = () => {
    setToken('');
    dispatch(clearLogin());
    history.push('/');
  };

  const activeTodos = todos?.reduce(function(accum, todo) {
    return isTodoCompleted(todo) ? accum : accum + 1;
  }, 0);

  return (
    <>
      <TodoPageWrapper className='d-flex d-justify-center d-align-center'>
        <div className='Todo__container'>
          <div className='Todo__creation'>
            <input
              ref={inputRef}
              className='Todo__input'
              placeholder='What need to be done?'
              onKeyDown={onCreateTodo}
            />
          </div>
          <div className='Todo__list'>
            {todos.length > 0 ? (
              <>
                <input
                  type='checkbox'
                  checked={activeTodos === 0}
                  onChange={onToggleAllTodo}
                  className='checkall'
                />
              </>
            ) : (
              <div />
            )}
            {todos.map(todo => (
              <SingleTodo
                todo={todo}
                key={todo.id}
                onUpdateStatus={handleUpdateStatus}
                onDeleteTodo={handleDeleteTodo}
              />
            ))}
          </div>
          <div className='Todo__toolbar'>
            <div className='Todo__tabs'>
              <BaseButton onClick={handleActiveAll}>Active</BaseButton>
              <BaseButton onClick={handleCompleteAll}>Completed</BaseButton>
              <BaseButton danger onClick={handleDeleteAll}>
                Clear all todos
              </BaseButton>
            </div>
          </div>
          <div className='d-flex d-justify-center d-align-center'>
            <BaseButton className='btn__logout' onClick={handleLogout}>
              {' '}
              Logout{' '}
            </BaseButton>
          </div>
        </div>
      </TodoPageWrapper>
    </>
  );
};

export default ToDoPage;
