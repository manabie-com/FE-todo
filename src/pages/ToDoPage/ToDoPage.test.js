import React from 'react';
import { mount } from 'enzyme';

import ToDoPage from './ToDoPage';

describe('render ToDoPage', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const mockProps = {
    todos: [{}, {}, {}],
  }

  const Component = mount(<ToDoPage {...mockProps} />);

  it('should render', () => {
    expect(Component.length).toBe(1);
  });
  it('match snapshot', () => {
    expect(Component).toMatchSnapshot();
  });

  it('should have todos data', () => {
    expect(mockProps.todos.length).toEqual(3);
  });
});
