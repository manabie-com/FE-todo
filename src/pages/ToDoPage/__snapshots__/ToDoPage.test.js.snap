// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ToDoPage match snapshot 1`] = `ReactWrapper {}`;
